kroxylicious-kms-providers/kroxylicious-kms-provider-azure-key-vault-kms/src/main/java/io/kroxylicious/kms/provider/azure/keyvault/SupportedKeyType.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.azure.keyvault;
+
+import java.util.Optional;
+
+public enum SupportedKeyType {
```

Comment:
this is a way of classifying some key types as there are `RSA` and `RSA-HSM` variants that aren't really different from our point of view, they will use the same wrap algorithm.

But I'm wondering if it would be better to keep it closer to what Azure exposes. So have a `KeyType` enum and an algorithm enum that matches Azure and operate on those.